### PR TITLE
ConDec-510: Refactor GitClient implementation so that it uses git repository file system management

### PR DESCRIPTION
**Pull request by lw2011**

### CODE REVIEW

**Comment by lw2011** [comment]
When this branch can be merged with branch in PR #137, PR's #137 branch will be rebased to most current develop branch and then the code review of #137 can be started.

@dekome, @kleebaum  could you review/comment on this branch?

File: src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/impl/GitClientImpl.java [hunk 1]
```diff
@@ -284,12 +363,32 @@ private static void deleteFolder(File directory) {
 	@Override
 	public List<RevCommit> getCommits() {
 		List<RevCommit> commits = new ArrayList<RevCommit>();
-		for (Ref branch : getAllRefs()) {
+		for (Ref branch : getOnlyRemoteRefs()) {
+			/* @issue: All branches will be created in separate file system
+			 * folders for this method's loop. How can this be prevented?
+			 *
+			 * @alternative: remove this method completely,
+			 * fetching commits from all branches is not sensible!
```
Comment (kleebaum) [inline src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/impl/GitClientImpl.java]: Yes, I think it's a bad idea to not fetch all commits of all branches. 
[alternative] We could check wether the JIRA issue key is part of the branch name and - if so - only use the commits from this branch. [/alternative]

File: src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/impl/GitClientImpl.java [hunk 2]
```diff
@@ -306,33 +405,78 @@ private Ref getRef(String jiraIssueKey) {
...
 		}
+		switchGitClientBackToDefaultDirectory();
 		return commits;
 	}
 
+	private void switchGitClientBackToDefaultDirectory() {
+		File directory = new File(fsManager.getDefaultBranchPath());
+		try {
+			git.close();
+			git=git.open(directory);
```
Comment (kleebaum) [inline src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/impl/GitClientImpl.java]: We should have a guideline on how to format our code. I always insert spaces befor and after an equal sign "=" like above. @lw2011 and @dekome , do you agree on inserting more spaces?

File: src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/impl/GitClientImpl.java (hunk 1 above)
Comment (lw2011) [reply src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/impl/GitClientImpl.java]: Thanks for your thoughts, I wil add the alternative to the code.

**Comment by lw2011** [comment]
Closing this PR, new PR will be started compared to develop branch, because "CONDEC-498.Git.repository.filesystem.management" is already part of develop.

**Pull request closed without merging**